stream-video-android-core/src/test/kotlin/io/getstream/video/android/core/base/IntegrationTestBase.kt

```diff
@@ -59,25 +61,28 @@ object IntegrationTestState {
     var call: Call? = null
 }
 
-open class IntegrationTestBase(connectCoordinatorWS: Boolean = true) : TestBase() {
+open class IntegrationTestBase(val connectCoordinatorWS: Boolean = true) : TestBase() {
     /** Client */
-    val client: StreamVideo
+    lateinit var client: StreamVideo
 
     /** Implementation of the client for more access to interals */
-    internal val clientImpl: StreamVideoImpl
+    internal lateinit var clientImpl: StreamVideoImpl
 
     /** Tracks all events received by the client during a test */
-    var events: MutableList<VideoEvent>
+    lateinit var events: MutableList<VideoEvent>
 
     /** The builder used for creating the client */
-    val builder: StreamVideoBuilder
+    lateinit var builder: StreamVideoBuilder
 
     var nextEventContinuation: Continuation<VideoEvent>? = null
     var nextEventCompleted: Boolean = false
 
     init {
         MockKAnnotations.init(this, relaxUnitFun = true)
+    }
 
+    @Before
+    fun setupVideo() {
```

Comment:
I moved all this code deliberately from `init { ... }` to `@Before fun ...` because the test dispatcher was set too late in the `TestBase`, the `init {...}` was already finished by that time and it caused issues (e.g. the `StreamVideoImpl.scope` was a real Dispatchers.IO and not a test dispatcher.

Reply:
This is a good point.